.eslintrc.json

```diff
@@ -3,18 +3,26 @@
     "browser": true,
     "es2021": true
   },
-  "extends": ["eslint:recommended", "plugin:react/recommended"],
+  "extends": [
+    "eslint:recommended",
+    "plugin:react/recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended"
+  ],
+  "parser": "@typescript-eslint/parser",
   "parserOptions": {
     "ecmaFeatures": {
       "jsx": true
     },
     "ecmaVersion": 2021,
     "sourceType": "module"
   },
-  "plugins": ["react"],
+  "plugins": ["react", "@typescript-eslint"],
   "rules": {
-    "no-unused-vars": "warn",
+    "no-unused-vars": "off",
+    "@typescript-eslint/no-unused-vars": ["warn"],
     "no-case-declarations": "warn",
-    "react/prop-types": "warn"
+    "react/prop-types": "off",
+    "react/react-in-jsx-scope": "off"
```

Comment:
The prop-types rule isnt needed any more with typescript so I've turned it off. The second rule has been added to disable a "React must be in scope when using jsx react" message. This message says you need to import React from 'react' but for functional components, this is no longer required. See https://stackoverflow.com/questions/42640636/react-must-be-in-scope-when-using-jsx-react-react-in-jsx-scope